### ISSUE TITLE
pynets thresholding

### ISSUE THREAD

**Issue by Aimanyounas** [dPys/PyNets#556]
if i import the pynet library version 1.27 i am not able to get the pynet.thresholding.threshold_propotional(matrix, threh_value) working, it says "AttributeError: module 'pynets' has no attribute 'thresholding'"

**Comment by dPys** [comment]
Hi @Aimanyounas -- looks like you might be using the wrong import?

Try:
```
from pynets.core.thresholding import threshold_proportional 
```

**Comment by Aimanyounas** [comment]
Like for my one matrix after applying threshold if i use *np.size(np.where(matrix))* i am getting such values, What should i conclude from it. My data is an rs-fmri images.
Subject  :  0
***thresh1=0.75***:  229908
****thresh2=0.85****:  260564
****thres3=0.5***:  153272
****thres4=0.25***:  76636
****thres5=0.125***:  38320
****thres4=0.95***:  291216

**Comment by dPys** [comment]
Hi @Aimanyounas -- these are good questions, and cut to the heart of the problem with analyzing functional brain graphs in general. There is no single universal threshold that has been recommended in this context (or any). So much of the answer depends on what _you_ as the researcher define to be "strong".  I'd personally suggest bootstrapping whatever analysis that you are performing across a variety of thresholds to see how your results fluctuate (and they will).  But then of course the issue becomes _which_ range of thresholds is appropriate -- i.e. the arbitrariness cannot be entirely avoided, even if its impact can be minimized. I'd recommend reading the classic 2010 paper on this: https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0013701 and more recently: https://pubmed.ncbi.nlm.nih.gov/28167349/

@dPys

**Comment by Aimanyounas** [comment]
I will have read of the papers for the better understanding, thanks a lot for being so helpful and responsive. My aim is to apply threshold on a connectivity matrix to reduce the number of connections( to only choose the significant connections). So now as you know my issue kindly impart your wisdom accordingly.

**Comment by dPys** [comment]
Again, what do you mean by "significant"?

**Comment by Aimanyounas** [comment]
Well i have a connectivity matrix that contains false positive. The functional connectivity matrix is typically estimated using the correlation between time series, where each cell in the connectivity matrix contains a correlation coefficient which crosses of the correlation between two regions of the brain. Therefore, i need to use proportional thresholding(or the one you will suggest) of the functional connectivity matrix to reduce the number of edges and select only the significant edges using PyNets module.

**Comment by dPys** [comment]
@Aimanyounas -- to clarify, I understand completely what you're trying to do, but what I'm getting at is more of a statistical/conceptual question. To put this another way, what do you define as being a "false positive"? Is it a total lack of connectivity between two brain regions? If so, how can we know that it is truly "false" without first having some ground-truth to compare the connectivity matrix against. One might even say that two regions are only "significantly" correlated over time in one frequency band (e.g. 0.028 Hz) but not in another (0.08 Hz). Fitting a correlation estimator to your time-series will result in a fully-connected connectivity matrix every time, even if the _weights_ of each individual connection might vary.

In other words, there is no strict definition of what counts as significant. Just as p<0.05 (or p<0.01) is more or less arbitrary in hypothesis testing, proportional thresholds are arbitrary in functional connectomics. You might ask then-- why threshold at all? Well, the answer is that it's helpful to see how your statistical inferences about a connectome might vary as a function of varying constraints on the level of sparsity that one may care about for a particular use-case.

**Comment by Aimanyounas** [comment]
Thanks a bunch for your wisdom. It really means a lot. Could you kindly further guide how to get the strong edges in a connectivity matrix? Is there libraries that are already there doing such work or should make my own function?

**Comment by dPys** [comment]
@Aimanyounas 
I'm beginning to get the sense that we might be going around in circles trying to address your question, but I also still want to be helpful just the same. Since I think your question is actually more of a theoretical one that is beyond the scope of this package specifically, I'm going to refer you to van den Heuvel's article which lays out some recommendations for exactly what you're trying to accomplish: https://www.sciencedirect.com/science/article/pii/S105381191730109X. To that end, `threshold_proportional` should be all you need to follow those recommendations.

Hope this helps,
@dPys

**Comment by dPys** [comment]
Closing this issue. Let me know if you'd like me to reopen.
